1.. create a sample django app that has some api end points 

 a))creating addUser
   ##note JsonResponse is imported from django.http
          JsonResponse needs data(dict) to be returned 

def addUser(request):
    val={
        'response':'User Added'

    }
    print("hello addUser")
    status=200
    return JsonResponse(val,status=status)


###Jsonrespon_le_pani_afai_display_hune_rahecha_bothin_page_and_console
===>after yo if we run the server and go to this Url 
first)in the web page
     {"response": "User Added"}

second) in the console
         hello addUser
        [29/Sep/2021 00:01:50] "GET /addUser HTTP/1.1" 200 26


b))addSomething

c)) addNew/<somerandomnum> 
##note status 500 is Internal server error

this returns 500 if some random num is > 50


d))addNewError
here, we make an actual exceptional error and sending the traceback into my front command


##so basically internal server error without try and except crashes the program 
   but the use of try and except, the except handles the exception that is it prints 
   what is wrong without crashing the prog.``



2...
write a notebook to test api 


##note practise of requests (import requests
				url='http://localhost:8000/'
				res=requests.get(url)
					or
				res=requests.get(url+"admin")
				res.text
					
				)


                                 
===>while calling the api 




















 
